src/theme/src/default-theme/utils/memoizeClassName.js

```diff
@@ -27,7 +40,9 @@ const memoizeClassName = fn => {
   // Return the wrapped function.
   return (...args) => {
     // Create a key by joining all args.
-    const key = args.join('_') || '__no_args__'
+    const key =
+      args.map(o => (typeof o === 'object' ? idOf(o) : o)).join('_') ||
```

Comment:
We need to be able to tell `theme` objects apart in instances where we have a nested theme. This happens because all of our `get*` functions are memoized. For example, `const getLinkAppearance = (color, theme) => {...}` (I'll respond more to passing `theme` in a separate comment, but let's assume it's required for this comment) Because `theme.palette[color].base` requires the theme object, we need to ensure we have the same theme before relying on the memoized value.

By default, all Objects coerce to the string `[Object object]` which would break memoization.

We could (in theory) add a property to the theme object that allows us to uniquely identify each theme in use. However, this gives us forward compatibility should the `get*` methods ever take additional object arguments, and it would be trivial to add array support if the need arises.